### PR TITLE
feat: enable reward model training with multiple validation datasets

### PR DESCRIPTION
**Pull request by jveronvialard**
# What does this PR do ?

This PR enables reward model training with multiple validation datasets.

# Usage

In addition to a main validation dataset, you can optionally specify additional validation datasets in data_prefix. The additional keys should start with "validation". Metrics are computed for each validation dataset and logged in separate wandb tabs.

Before:
```python
python ${NEMO_ALIGNER_PATH}/examples/nlp/gpt/train_reward_model.py \
    ...
    ++model.data.data_prefix="{train:[${TRAIN_DATA}],validation:[${VAL_DATA}],test:[${TEST_DATA}]}" \
    ...
```

After:
```python
python ${NEMO_ALIGNER_PATH}/examples/nlp/gpt/train_reward_model.py \
    ...
    ++model.data.data_prefix="{train:[${TRAIN_DATA}],validation:[${VAL_DATA}],validation_other:[${VAL_OTHER_DATA}],test:[${TEST_DATA}]}" \
    ...
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation? Make sure to also update the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/index.html) which contains the tutorials

# Checklist when contributing a new algorithm
- [x] Does the trainer resume and restore model state all states?
- [x] Does the trainer support all parallelism techniques(PP, TP, DP)?
- [x] Does the trainer support `max_steps=-1` and `validation`?
- [x] Does the trainer only call APIs defined in [alignable_interface.py](/nemo_aligner/models/alignable_interface.py)?
- [x] Does the trainer have proper logging?

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by odelalleau — COMMENTED** [review]
Thanks! Couple of minor comments -- also can you please add a new feature entry to the CHANGELOG?

File: examples/nlp/gpt/conf/training_rm.yaml [hunk 1]
```diff
@@ -113,7 +113,8 @@ model:
     reset_attention_mask: False # Reset attention mask after end-of-document token
     eod_mask_loss: False # Mask loss for the end of document tokens
     index_mapping_dir: null # path to save index mapping .npy files, by default will save in the same location as data_prefix
-    data_prefix: null
+    data_prefix: null # For multiple validation sets, use a dictionary that includes one key named `validation`
+    # (main validation set) and extra keys of the form `validation*` (secondary validation sets).
```
Comment (odelalleau) [inline examples/nlp/gpt/conf/training_rm.yaml:117]: This format of comment looks a bit weird to me (continuing an inline comment on the next line) => suggestion
```suggestion

    # For multiple validation sets, use a dictionary that includes one key named `validation` (main
    # validation set) and extra keys of the form `validation*` (secondary validation sets), for instance:
    # model.data.data_prefix="{train:[train.jsonl],validation:[valid.jsonl],validation_set2:[valid2.jsonl],test:[test.jsonl]}"
    data_prefix: null 
```

**Review by odelalleau — COMMENTED** [review]
One last small comment fix and we're good to go!

**Review by odelalleau — APPROVED** [review]
Good to go, thanks!
